precompile/contracts/feemanager/contract.go

```diff
@@ -242,6 +249,34 @@ func setFeeConfig(accessibleState contract.AccessibleState, caller common.Addres
 		return nil, remainingGas, fmt.Errorf("%w: %s", ErrCannotChangeFee, caller)
 	}
 
+	// Add a log to be handled if this action is finalized.
+	if contract.IsDUpgradeActivated(accessibleState) {
+		if remainingGas, err = contract.DeductGas(remainingGas, GetFeeConfigGasCost+FeeConfigChangedEventBaseGasCost); err != nil {
+			return nil, 0, err
+		}
+		oldConfig := GetStoredFeeConfig(stateDB)
+		topics, data, err := PackFeeConfigChangedEvent(
+			caller,
+			oldConfig,
+			feeConfig,
+		)
+		if err != nil {
+			return nil, remainingGas, err
+		}
+
+		logDataGasCost := uint64(len(data)) * contract.LogDataGas
```

Comment:
Could we subtract `GetFeeConfigGasCost` at the top here and move the entire calculation of the gas cost due to the produced log into one place, so that it's not split up?

Reply:
moved